docs/source/example_notebooks/gcm_linear_scm_bootstap_pvalues/false_discovery_rate_control.py

```diff
@@ -0,0 +1,152 @@
+"""Functions in this module should be considered experimental, meaning there might be breaking API changes in the
+future.
+"""
+
+from typing import Union, List, Optional, Callable
+from scipy.stats import rankdata
+import numpy as np 
+
+def numberfy(x):
+    ''' if x could be a number return it as numeric even if it is in string format '''
+    try:
+        nx=float(x)
+    except:
+        nx=np.NaN
+    return(nx)
+
+def adjusted_FDR_pvalues(unajusted_p_values: Union[np.ndarray, List[float]]) -> np.ndarray:
+                        
+    """ adjusts p-values for FDR using the standard Benjamini-Hochberg (Stepup) procedure
+    
+    Benjamini Y., Hochberg Y. (1995). Controlling the false discovery rate: a practical and powerful approach to multiple testing, Journal of the Royal Statistical Society B, 289-300.
+
+    Note: one can use 
+     import statsmodels.stats.multitest as smt
+     adjusted_pvalues = smt.multipletests(pvals=(0.05, 0.01, 0.1), method='fdr_bh', is_sorted=False, returnsorted=False)[1]
+     however, this does not allow for missing values which could occure is using semopy to estimate an SEM with a latent node
+
+    :param unajusted_p_values: An array or a list of p-values. [could include NANs]
+    
+    :return: Adjusted FDR p-Values in same order as the input pvaules. 
+
+    """
+
+    mp=len(unajusted_p_values)
+    p_valsArr = np.array(unajusted_p_values)
+
+    #before starting ensure numeric values and just count them - i.e. exlude null values
+    m=0
+    p_vals = np.empty(mp)
+    for i in range(mp):
+        p_vals[i]=numberfy(p_valsArr[i])
+        if p_vals[i] >= 0:
+            m=m+1
+        
+    #calcualte FDR adjusted pvalues
+    ranked_p_values = rankdata(p_vals)
+    adjfdr = p_vals * m / ranked_p_values
+    adjfdr[adjfdr > 1] = 1
+    
+    #rememebr if (k) is rejected (1)..(k) are rejected - the FDR adjustemnt shoud reflect that
+    IndexRanks = p_vals.argsort()
+    minAP=1
+    for i in range(mp-1,-1,-1):
+        pointer = int(IndexRanks[i])
+        if adjfdr[pointer] > minAP:
+            adjfdr [pointer] = minAP
+        else:
+            minAP = adjfdr[pointer]
+    return adjfdr
+
+
+def adjusted_FDCR_pvalues(unajusted_p_values,
+                          belife_scores,
+                          intersection_belife_score=10000.0
```

Comment:
Also specify the parameter types, e.g. `believe_scores: Union[np.ndarray, List[float]]` or `intersection_believe_score: int = 10000.0`